### PR TITLE
DB-424 Force extension signature check on startup

### PR DESCRIPTION
**Pull request by chrmod**

### CODE REVIEW

File: cliqz.cfg [hunk 1]
```diff
@@ -18,6 +21,25 @@ let extensionUrls = {
 
 let prefs = Cc["@mozilla.org/preferences-service;1"].getService(Ci.nsIPrefBranch);
 
+function verifyAddons() {
+  Cu.import("resource://gre/modules/addons/XPIProvider.jsm");
+  let versionChecker = Cc["@mozilla.org/xpcom/version-comparator;1"]
+                         .getService(Components.interfaces.nsIVersionComparator);
```
Comment (maxim-cliqz) [inline cliqz.cfg]: Weird (and wasteful) indentation

File: cliqz.cfg [hunk 2]
```diff
@@ -18,6 +21,25 @@ let extensionUrls = {
...
+  let distributionVersion = prefs.getCharPref("distribution.version");
+
+  let lastDistributionVersion;
+  try {
```
Comment (maxim-cliqz) [inline cliqz.cfg]: Can it really throw?

File: cliqz.cfg [hunk 3]
```diff
@@ -18,6 +21,25 @@ let extensionUrls = {
...
+    lastDistributionVersion = prefs.getCharPref("distribution.lastVersion");
+  } catch(e) {
+    lastDistributionVersion = 0;
+  }
+
+  if (versionChecker.compare(distributionVersion, lastDistributionVersion) > 0) {
+    XPIProvider.verifySignatures();
+    prefs.setCharPref("distribution.lastVersion", distributionVersion);
```
Comment (maxim-cliqz) [inline cliqz.cfg]: There is "extensions.lastPlatformVersion" which serves the same purpose and is already maintained by `AddonManager`.
And actually I think it's best to do it where all other similar job is done – in `AddonManagerInternal.startup`

Comment (luciancor) [inline cliqz.cfg]: temporary solution which should be removed/changed with a better one soon

File: cliqz.cfg (hunk 2 above)
Comment (luciancor) [inline cliqz.cfg]: the pref API is quite strange - if the pref is not there or not ready it simply throws an exception :/

**Pull request merged**